src/Entity.js

```diff
@@ -60,6 +64,11 @@ export class Entity {
 
   getMutableComponent(Component) {
     var component = this._components[Component._typeId];
+
+    if (!component) {
+      return;
+    }
+
```

Comment:
No point going through queries if the component does not exist.

Reply:
nice catch yep